api/token_metadata.cc

```diff
@@ -21,6 +21,9 @@ using namespace json;
 void set_token_metadata(http_context& ctx, routes& r, sharded<locator::shared_token_metadata>& tm) {
     ss::local_hostid.set(r, [&tm](std::unique_ptr<http::request> req) {
         auto id = tm.local().get()->get_my_id();
+        if (!bool(id)) {
+            throw not_found_exception("not yet set");
```

Comment:
please extend this to include *what* is not yet set

Reply:
Well, the caller queries /storage_service/hostid/local, so it should be clear what's not yet set :smile: 
On the other hand, if put into logs it's likely that this message will go on its own, so yes, worth being more verbose

Reply:
Yes, this is clear enough in the code, I was thinking ahead of the poor soul seeing this in the logs and wandering what it could mean.